src/transaction/components/TransactionProvider.tsx

```diff
@@ -82,12 +108,62 @@ export function TransactionProvider({
     setLifeCycleStatus,
     transactionHashList,
   });
-  const { transactionHash, status: callStatus } = useCallsStatus({
+  // useSendCalls or useSendCall
+  // Used for contract calls with raw calldata.
+  const { status: statusSendCalls, sendCallsAsync } = useSendCalls({
     setLifeCycleStatus,
-    transactionId,
+    setTransactionId,
+  });
+  const {
+    status: statusSendCall,
+    sendCallAsync,
+    data: sendCallTransactionHash,
+  } = useSendCall({
+    setLifeCycleStatus,
+    transactionHashList,
   });
+
+  // Transaction type and status
+  // Returns the appropriate transaction type and status based on the provided props and wallet capabilities.
+  const { transactionType, transactionStatus } = useTransactionType({
+    calls,
+    contracts,
+    transactionStatuses: {
```

Comment:
I understand the reason behind this, one question here is, I feel like we should we set this as default values. And avoid passing them here.

Reply:
Just to keep the code more chill

Reply:
would it be possible to move all these calls/contracts hooks into a hook and to just be able to call useTransaction with calls/contracts?